test/test-pipe-getsockname.c

```diff
@@ -162,47 +162,48 @@ TEST_IMPL(pipe_getsockname) {
 
 
 TEST_IMPL(pipe_getsockname_abstract) {
+  /* TODO(bnoordhuis) Use unique name, susceptible to concurrent test runs. */
+  static const char name[] = "\0" TEST_PIPENAME;
 #if defined(__linux__)
-  char buf[1024];
-  size_t len;
-  int r;
-  int sock;
-  struct sockaddr_un sun;
-  socklen_t sun_len;
-  char abstract_pipe[] = "\0test-pipe";
-
-  sock = socket(AF_UNIX, SOCK_STREAM, 0);
-  ASSERT(sock != -1);
-
-  sun_len = sizeof sun;
-  memset(&sun, 0, sun_len);
-  sun.sun_family = AF_UNIX;
-  memcpy(sun.sun_path, abstract_pipe, sizeof abstract_pipe);
-
-  r = bind(sock, (struct sockaddr*)&sun, sun_len);
-  ASSERT(r == 0);
-
-  r = uv_pipe_init(uv_default_loop(), &pipe_server, 0);
-  ASSERT(r == 0);
-  r = uv_pipe_open(&pipe_server, sock);
-  ASSERT(r == 0);
-
-  len = sizeof buf;
-  r = uv_pipe_getsockname(&pipe_server, buf, &len);
-  ASSERT(r == 0);
-
-  ASSERT(memcmp(buf, abstract_pipe, sizeof abstract_pipe) == 0);
-
-  uv_close((uv_handle_t*)&pipe_server, pipe_close_cb);
-
-  uv_run(uv_default_loop(), UV_RUN_DEFAULT);
-
-  close(sock);
-
-  ASSERT(pipe_close_cb_called == 1);
+  char buf[256];
+  size_t buflen;
+
+  buflen = sizeof(buf);
+  memset(buf, 0, sizeof(buf));
```

Comment:
```suggestion
  memset(buf, 0, buflen);
```

Reply:
I personally prefer the current approach because it doesn't require context from surrounding lines.

Also, I was raised in a time when compilers were supremely bad at constant propagation so I have a really hard time of letting go of behaviors like ☝️